### PR TITLE
[do not merge] selfcancelxfer - better ts code

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
@samuelsov @mlutfy  https://github.com/civicrm/civicrm-core/pull/29408 addresses the regression that @samuelsov reported but it seems to me we shoud do something more like this in master.... - probably we should go a step further & handle plural better but I can't imaging the way the before vs after is being done is much fun on translators

### CODE REVIEW

File: xml/templates/message_templates/event_online_receipt_html.tpl [hunk 1]
```diff
@@ -461,9 +461,11 @@
       {if {event.allow_selfcancelxfer|boolean}}
         <tr>
           <td colspan="2" {$valueStyle}>
-            {capture assign=selfservice_preposition}{if {event.selfcancelxfer_time} > 0}{ts}before{/ts}{else}{ts}after{/ts}{/if}{/capture}
-            {ts 1={event.selfcancelxfer_time} 2=$selfservice_preposition}You may transfer your registration to another participant or cancel your registration up to %1 hours %2 the event.{/ts}
-            {if {contribution.paid_amount|boolean}}{ts}Cancellations are not refundable.{/ts}{/if}<br/>
+            {if {event.selfcancelxfer_time|boolean} && {event.selfcancelxfer_time} > 0}
+              {ts 1="{event.selfcancelxfer_time}"}You may transfer your registration to another participant or cancel your registration up to %1 hours before the event.{/ts}
+            {else}
```
Comment (seamuslee001) [inline xml/templates/message_templates/event_online_receipt_html.tpl:466]: @eileenmcnaughton just wondering if this should be elseif with the |boolean option as what if there is no selfcancelxfer time set?

Also we seem to have lost the statement from old like 466

Comment (eileenmcnaughton) [reply xml/templates/message_templates/event_online_receipt_html.tpl:466]: @seamuslee001 yeah - I thought that too - I will wait for @mlutfy & @samuelsov to take a look cos they might have further thoughts

on losing that line - I'm a bit confused about that - I was sure I had already removed that line on the basis of ... who are we to say what is refundable!

**Comment by axelady** [comment]
See comments on https://lab.civicrm.org/dev/core/-/issues/5006 which was not resolved by upgrading to 5.70.1  The proposed change by https://lab.civicrm.org/BobS does fix it.

**Comment by demeritcowboy** [comment]
This has conflicts and another PR was merged and this says do not merge, so, not merging...

**Comment by eileenmcnaughton** [comment]
thanks for the clean up @demeritcowboy  - I think I was wanting to merge stuff to the rc first originally  - but it does seem worth getting rid of assign+interpolate in favour of using ts correctly. OTOH there wasn't much interest in this one & I don't want to do a PR if there is no real interest - esp since I'm enjoying seeing the number of open PRs fall....

**Pull request closed without merging**